tests/apollo/util/bft.py

```diff
@@ -1105,42 +1101,53 @@ async def wait_for(self, predicate, timeout, interval):
             with trio.fail_after(timeout):
                 while True:
                     with trio.move_on_after(interval):
-                        if await predicate():
-                            return
-                        await trio.sleep(0.1)
+                        if inspect.iscoroutinefunction(predicate):
+                            result, status = await predicate()
+                            if status:
+                                return result
+                            else:
+                                await trio.sleep(0.1)
+                        else:
+                            raise TypeError
+
+    async def retrieve_metric(self, replica_id, component_name, type_, key):
+        try:
+            return await self.metrics.get(replica_id, component_name, type_, key)
+        except KeyError:
+            return None    
```

Comment:
We should probably show an error message in case someone puts in a wrong metric name for convenience.